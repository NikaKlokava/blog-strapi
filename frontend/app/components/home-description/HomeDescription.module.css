.main_description {
  padding-left: 50px;
  padding-right: 50px;
  background-color: #e0b8c64d;
}

.title {
  font-size: 30px;
  align-self: center;
}

.background_img {
  min-height: 100vh;
  width: 100vw;
  filter: opacity(50%);
}

@media screen and (max-width: 550px) {
  .background_img {
    min-height: 40vh;
    width: 100%;
    background-size: cover;
  }
}