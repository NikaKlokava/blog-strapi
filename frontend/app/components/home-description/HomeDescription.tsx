import { getData } from "@/app/utils/utils";
import { homeMockData } from "@/mocks/mocks";
import Image from "next/image";
import styles from "./HomeDescription.module.css";

export const HomeDescription = async () => {
  const data: HomeData = await getData("homes");
  return (
    <>
      <div className={styles.main_description}>
        {(data || homeMockData).attributes.description}
      </div>
      <div className={styles.title}>
        {(data || homeMockData).attributes.title}
      </div>
      <Image
        src={
          data
            ? process.env.STRAPI_API_URL +
              data?.attributes.background.data.attributes.url
            : homeMockData.attributes.background
        }
        alt={"background_img"}
        priority={true}
        width={1920}
        height={1080}
        style={{
          objectFit: "cover",
        }}
        sizes={"(max-width: 600px) 100vw, (max-width: 400px) 100vw, 100vw"}
        className={styles.background_img}
      />
    </>
  );
};
