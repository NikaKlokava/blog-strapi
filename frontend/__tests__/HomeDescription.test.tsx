import { HomeDescription } from "@/app/components/home-description/HomeDescription";
import { render } from "@testing-library/react";

describe("Test the HomeDescription component", () => {
  it("The HomeDescription component renders correctly", async () => {
    const homeDescriptionSnap = render(
      await (async () => await HomeDescription())()
    );
    expect(homeDescriptionSnap).toMatchSnapshot();
  });
});
