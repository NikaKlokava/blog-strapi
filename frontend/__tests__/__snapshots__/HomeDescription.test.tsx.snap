// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the HomeDescription component The HomeDescription component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_description"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Temporibus
  fugit, aliquam animi nostrum repudiandae non quibusdam. Commodi,
  blanditiis, mollitia eaque reprehenderit fugiat, aut facere dolorem
  voluptatum a nulla ut. Iure? Lorem ipsum dolor sit amet consectetur
  adipisicing elit. Temporibus fugit, aliquam animi nostrum repudiandae
  non quibusdam. Commodi, blanditiis, mollitia eaque reprehenderit fugiat,
  aut facere dolorem voluptatum a nulla ut. Iure?
      </div>
      <div
        class="title"
      >
        Hello from TRAVELER!
      </div>
      <img
        alt="background_img"
        class="background_img"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="1080"
        sizes="(max-width: 600px) 100vw, (max-width: 400px) 100vw, 100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="color: transparent; object-fit: cover;"
        width="1920"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="main_description"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Temporibus
  fugit, aliquam animi nostrum repudiandae non quibusdam. Commodi,
  blanditiis, mollitia eaque reprehenderit fugiat, aut facere dolorem
  voluptatum a nulla ut. Iure? Lorem ipsum dolor sit amet consectetur
  adipisicing elit. Temporibus fugit, aliquam animi nostrum repudiandae
  non quibusdam. Commodi, blanditiis, mollitia eaque reprehenderit fugiat,
  aut facere dolorem voluptatum a nulla ut. Iure?
    </div>
    <div
      class="title"
    >
      Hello from TRAVELER!
    </div>
    <img
      alt="background_img"
      class="background_img"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="1080"
      sizes="(max-width: 600px) 100vw, (max-width: 400px) 100vw, 100vw"
      src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
      style="color: transparent; object-fit: cover;"
      width="1920"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
